src/DotNet.Testcontainers/Containers/Configurations/Misc/ResourceReaperContainerConfiguration.cs

```diff
@@ -0,0 +1,77 @@
+namespace DotNet.Testcontainers.Containers.Configurations.Misc
+{
+  using System;
+  using JetBrains.Annotations;
+  using WaitStrategies;
+
+  public class ResourceReaperContainerConfiguration
+  {
+    /// <summary>
+    /// See <see cref="Image"/>
+    /// </summary>
+    private const string DefaultImage = "ghcr.io/psanetra/ryuk:2021.12.20";
+
+    public ResourceReaperContainerConfiguration(string image = DefaultImage, Guid? sessionId = null, string name = null, int port = 0, int defaultPort = 8080)
+    {
+      this.SessionId = sessionId ?? Guid.NewGuid();
+      this.Name = name ?? $"testcontainers-ryuk-{this.SessionId:D}";
+      this.Image = image;
+      this.Port = port;
+      this.DefaultPort = defaultPort;
+      this.WaitStrategy = new WaitForContainerUnix().UntilPortIsAvailable(defaultPort);
+    }
+
+    /// <summary>
+    /// Gets or sets the reaper container image.
+    /// Must be compatible to ryuk and support pruning on SIGINT and SIGTERM signals.
+    /// As of ryuk 0.3.3 pruning is not supported by the original ryuk image. By default the fork ghcr.io/psanetra/ryuk:2021.12.20 is used.
```

Comment:
As you can see I am currently using a custom build of ryuk here. I think we should wait with merging this PR for the PR https://github.com/testcontainers/moby-ryuk/pull/22 to be merged, so that we can use the original ryuk image.